### ISSUE TITLE
Error running TF 0.6.16 in Alpine docker container: "no such file or directory"

### ISSUE THREAD

**Issue by ljfranklin** [hashicorp/terraform#6998]
I'm hitting an error running terraform 0.6.16 inside an alpine docker image. Same result when building from master. Here's the error:

```
$ docker run -it ljfranklin/terraform-resource-lite terraform -v
no such file or directory
FATA[0000] Error response from daemon: Cannot start container 4d616113d6d0882b929e4bacc3da09e564bd14834075717ae7e36dfbadcf0979: [8] System error: no such file or directory
```

Switching back to 0.6.15 works as expected:

```
$ docker run -it ljfranklin/terraform-resource-lite terraform -v
Terraform v0.6.15
```

Here's the entire Dockerfile:

```
FROM alpine:3.4

ENV TERRAFORM_VERSION=0.6.16

RUN apk update && \
    apk add ca-certificates unzip wget && \
    wget -O /tmp/terraform.zip https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
    unzip /tmp/terraform.zip -d /usr/local/bin && \
    rm -rf /tmp/terraform.zip
```

Any ideas?

Thanks!

**Comment by ljfranklin** [comment]
After a bit more digging, looks like the 0.6.16 binary is now dynamically linked and requires glibc.

terraform 0.6.16

```
$ file terraform
terraform: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), stripped

$ readelf -a terraform
Version needs section '.gnu.version_r' contains 2 entries:
 Addr: 0x000000000163ed20  Offset: 0x123ed20  Link: 13 (.dynstr)
  000000: Version: 1  File: libpthread.so.0  Cnt: 2
  0x0010:   Name: GLIBC_2.3.2  Flags: none  Version: 2
  0x0020:   Name: GLIBC_2.2.5  Flags: none  Version: 3
  0x0030: Version: 1  File: libc.so.6  Cnt: 1
  0x0040:   Name: GLIBC_2.2.5  Flags: none  Version: 4
```

terraform 0.6.15

```
$ file terraform
terraform: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, stripped

$ readelf -a terraform
There is no dynamic section in this file.
```

Is this dynamic linking behavior expected or an oversight?

**Comment by enieuw** [comment]
I had the same issue, should be fixed with the next release, see:
https://github.com/hashicorp/terraform/issues/6714

**Comment by siddharthist** [comment]
This can be closed, it is a duplicate of #6714
